JAVA-1206: log custom type columns in query logger.

olim7t opened the pull request.

olim7t committed:
JAVA-1206: handle custom column types in query logger

adutra commented on driver-core/src/main/java/com/datastax/driver/core/DataType.java:
```diff
@@ -900,16 +900,16 @@ public String getCustomTypeClassName() {
+//        public String format(Object value) {
+//            throw new InvalidTypeException("Cannot format values of custom types");
+//        }
+//
```
What is the reason why you commented out these 2 methods?

adutra replied:
Oh nevermind, it's because `QueryLogger` now calls `format()` on custom types.

olim7t replied:
Yes, actually I meant to remove them.

olim7t committed:
Open up QueryLogger API

Make constructors and most internal methods protected, so that the basic
implementations can be extended and customized in user code.
